src/Components/Forms/src/DataAnnotationsValidator.cs

```diff
@@ -13,6 +13,8 @@ public class DataAnnotationsValidator : ComponentBase, IDisposable
 
     [CascadingParameter] EditContext? CurrentEditContext { get; set; }
 
+    [Inject] private IServiceProvider ServiceProvider { get; set; } = default!;
```

Comment:
I'm in two minds about the nullability annotation here. I know the only way you can make it `null` is to use an obsolete API, but existing code may do that, and then newer code will be getting told a lie by the annotation.

Should we keep it as `IServiceProvider?` until the obsoleted API is actually removed?

Reply:
Oh I see that @pranavkm just suggested marking it as non-nullable, so perhaps that is an established pattern about obsoletion already. Just waiting for a comment from him to confirm.

Reply:
This one is a private property and our implementation always expects injected properties to be available. So annotating it as non-nullable seems correct. Does that sound right?

Reply:
You're quite right. Sorry for the confusion!